source/common/http/utility.cc

```diff
@@ -751,6 +751,23 @@ const std::string& Utility::getProtocolString(const Protocol protocol) {
   NOT_REACHED_GCOVR_EXCL_LINE;
 }
 
+const std::string Utility::buildOriginalUri(const Http::RequestHeaderMap& request_headers,
+                                            const uint32_t max_path_length) {
+  if (!request_headers.Path()) {
+    return "";
+  }
+  absl::string_view path(request_headers.EnvoyOriginalPath()
+                             ? request_headers.getEnvoyOriginalPathValue()
+                             : request_headers.getPathValue());
```

Comment:
will we have case "envoy-origin-path" exists but not "path"?

Reply:
It appears in the router config_impl, that if `envoy-origin-path` exists, it is set to the `path` value. So no, there should not be any possibility of envoy-origin-path existing but path not existing.

Reply:
That code in the router_config runs after headers make it to the router, so I don't think this will have been set at this point. On the other hand I think the check for Path is unnecessary because `getPathValue()` should return an empty string in that case. Digging into it some more, original-path is only set in for certain route actions, so there might not be a point in reading it at this point at all?

Reply:
Ping on this, I think we can simplify this logic